### PR TITLE
feat(Project): Add new endpoint for project's license clearing tree view

### PR DESCRIPTION
**Pull request by hoangnt2**
### Pre-condition:
Turn on feature by adding this property into /etc/sw360/sw360.properties: **enable.flexible.project.release.relationship=true**

Create a new project with dependency network:
- Url: http://localhost:8080/resource/api/projects/network
- Method: POST
- Body: 
```
{
    "name": "Test_PR",
    "dependencyNetwork": [
        {
            "releaseId": "${release_id_1_in_sw360}", // Replace this by an existing release id in sw360
            "releaseRelationship": "CONTAINED",
            "mainlineState": "OPEN",
            "comment": "",
            "createOn": "2024-07-05",
            "createBy": "admin@sw360.org",
            "releaseLink": [
                {
                    "releaseId": "${release_id_2_in_sw360}", // Replace this by an existing release id in sw360 (difference from above)
                    "releaseRelationship": "OPTIONAL",
                    "mainlineState": "MAINLINE",
                    "comment": "",
                    "createOn": "2024-07-05",
                    "createBy": "admin@sw360.org",
                    "releaseLink": []
                }
            ]
        }
    ]
}
```

### How To Test?
_**Replace {project_id} with the id of the project created in the pre-condition**_
#### Endpoint 1: Get directly linked resources of project (listing all directly linked projects and releases of a project)

- Url: http://localhost:8080/resource/api/projects/network/{project_id}/linkedResources
- Method: GET
- Expected output: The response body will be shown in below format:

```
{
    "id": "{project_id}",
    "name": "Test_PR",
    "relation": "UNKNOWN",
    "projectType": "PRODUCT",
    "state": "ACTIVE",
    "clearingState": "OPEN",
    "linkedReleases": [
        {
            "id": "{release_id_1_in_sw360}",
            "vendor": "",
            "name": "{release_1_name}",
            "version": "{release_1_version}",
            "longName": "{release_1_long_name}",
            "releaseRelationship": "CONTAINED",
            "mainlineState": "OPEN",
            "hasSubreleases": true,
            "clearingState": "NEW_CLEARING",
            "attachments": [],
            "componentType": "SERVICE",
            "licenseIds": [],
            "comment": "",
            "otherLicenseIds": [],
            "accessible": true,
            "index": 0,
            "projectId": "{project_id}",
            "releaseMainLineState": "OPEN"
        }
    ],
    "enableSvm": true
}
```

#### Endpoint 2: Get get indirect linked releases of a project in dependency network by release's index path (support fetch data on expand node from GUI)

- Url: http://localhost:8080/resource/api/projects/network/{project_id}/releases?path=0
- Method: GET
- Expected output: The response body will be shown in below format:

```
{
    "_embedded": {
        "sw360:releaseLinks": [
            {
                "id": "{release_id_2_in_sw360}",
                "vendor": "",
                "name": "{release_2_name}",
                "version": "{release_2_version}",
                "longName": "{release_2_long_name}",
                "releaseRelationship": "OPTIONAL",
                "mainlineState": "MAINLINE",
                "hasSubreleases": false,
                "clearingState": "NEW_CLEARING",
                "attachments": [],
                "componentType": "OSS",
                "licenseIds": [],
                "comment": "",
                "otherLicenseIds": [],
                "accessible": true,
                "index": 0,
                "projectId": "6d0c271a16ee457a8b608afc874e1dc9",
                "releaseMainLineState": "OPEN"
            }
        ]
    },
    "_links": {
        "curies": [
            {
                "href": "http://localhost:8080/resource/docs/{rel}.html",
                "name": "sw360",
                "templated": true
            }
        ]
    }
}
```

### CODE REVIEW

**Comment by sameed20** [comment]
Testing this PR.

**Comment by sameed20** [comment]
Test Successful
Testing_0
Testing_1
Testing_2

File: backend/common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 
 import org.apache.commons.io.IOUtils;
 import org.apache.commons.lang.StringUtils;
+import org.apache.jena.shared.NotFoundException;
```
Comment (GMishx) [inline backend/common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java:25]: Please remove unused import.

**Review by GMishx — CHANGES_REQUESTED** [review]
One minor change needed.

File: backend/common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java (hunk 1 above)
Comment (hoangnt2) [reply backend/common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java]: I removed it

**Pull request merged**